code/datums/martial/monk.dm

```diff
@@ -0,0 +1,455 @@
+			quivering_palm.Grant(owner)
+			quivering_palm.martial = src
+			available_actions = list(stunning_fist, wholeness_of_body, quivering_palm)
+		if(16)
+			to_chat(owner, "<span class = 'danger'>Your Ki has fully developed, and can be used to nullify armor.</span>")
+			ki_level = 100
+		if(17)
+			to_chat(owner, "<span class = 'danger'>You can perform more Stunning Fists before resting.</span>")
+			stunning_fist.max_uses = 14
+		if(20)
+			if(!is_disciple)
+				to_chat(owner, "<span class = 'danger'>You can now make your first Disciple.</span>")
+				make_disciple.Grant(owner)
+				make_disciple.uses++
+		if(25)
+			if(!is_disciple)
+				to_chat(owner, "<span class = 'danger'>You can now make your second Disciple.</span>")
+				make_disciple.uses++
+		if(30)
+			if(!is_disciple)
+				to_chat(owner, "<span class = 'danger'>You can now make your final Disciple.</span>")
+				make_disciple.uses++
+
+/datum/martial_art/monk/proc/attack_roll(mob/living/T, abm)
+	if(istype(T, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = T
+		if(H.mind && istype(H.mind.martial_art, /datum/martial_art/monk))
+			var/datum/martial_art/monk/M = H.mind.martial_art
+			return M.defense_roll(curr_ab + abm)
+	var/armor_class = 10 + (T.getarmor("chest", "melee") * 0.1)
+	var/attack_bonus = curr_ab + abm
+	if(T.incapacitated())
+		attack_bonus += 3
+	var/attack_roll = roll(1,20) + attack_bonus
+	if(attack_roll >= armor_class)
+		if(attack_roll >= 20)
+			attack_roll = roll(1,20) + attack_bonus
+			if(attack_roll >= armor_class)
+				return ATK_CRIT_HIT
+			return ATK_HIT
+		else
+			return ATK_HIT
+	return ATK_MISS
+
+/datum/martial_art/monk/proc/defense_roll(abm)
+	var/armor = (owner.getarmor("chest", "melee") * 0.1)
+	var/armor_class = 10
+	if(!armor)
+		armor_class = 15
+	else
+		armor_class -= armor
+	var/attack_bonus = abm
+	if(owner.incapacitated())
+		attack_bonus += 3
+	var/attack_roll = roll(1,20) + attack_bonus
+	if(attack_roll >= armor_class)
+		if(attack_roll >= 20)
+			attack_roll = roll(1,20) + attack_bonus
+			if(attack_roll >= armor_class)
+				return ATK_CRIT_HIT
+			return ATK_HIT
+		else
+			return ATK_HIT
+	return ATK_MISS
+
+/datum/martial_art/monk/proc/add_exp(amt)
+	if(current_level == level_cap)
+		return
+	current_exp += amt
+	if(current_exp >= next_level_exp)
+		current_level++
+		var/next_level = current_level + 1
+		next_level_exp = next_level*100
+		do_level_up()
+		to_chat(owner, "<span class = 'danger'>You feel more confident in your powers.</span>")
+
+/datum/martial_art/monk/proc/do_attack(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D, use_cleave, use_circle, surrounding_mobs)
+	var/picked_hit_type = pick("punched")
+	switch(picked_hit_type)
+		if("punched")
+			A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	var/atr = attack_roll(D, flurry_penalty())
+	if(atr)
+		var/old_stat = D.stat
+		if(D.stat != DEAD && D.ckey && D != A)
+			add_exp(5)

```

Comment:
please make sure this check is updated with any additional "make sure this person is actually a player" changes made since the original PR

Reply:
Believe everything here is still in working order from what I can tell.